### ISSUE TITLE
Does perspective transform support source points outside the image dimensions

### ISSUE THREAD

**Issue by Msabih** [arraiyopensource/kornia#133]
Hi,

I was experimenting with homography transform and one issue, I was consistently getting different results as compared to opencv. in warp perspective where the calculation of 'M' was correct.

 Before I check my code or discuss more about this issue, could you please answer if the source points outside image dimensions are supported ? Because I have such "source" points which are supported in OpenCV but not in TGM ?

**Comment by edgarriba** [comment]
In theory yes, however the sampler will fill it with zeros by default but can play with the padding mode parameter from the sampler. Could you provide us a short snippet reflecting such case?

**Comment by Msabih** [comment]
> In theory yes, however the sampler will fill it with zeros by default but can play with the padding mode parameter from the sampler. Could you provide us a short snippet reflecting such case?

Okay, i have to upload a test image. Please download this npz image 
https://transfer.sh/F6Kny/tgm_testimg.npz

and then this code snippet to compare the output of opencv and tgm.

```
import cv2
import numpy as np
import matplotlib.pyplot as plt
import torch
import torchgeometry as tgm

npz_file = np.load("testimg.npz")
gt_img = npz_file['arr_0']
points_src = np.array([[425.66718, 89.],[-3292.292,256.6667],[4270.3945 ,256.6667], [ 0.,89.]], dtype=np.float32)
points_dst = np.array([[ 852., 0.],[-426.,240.],[ 852., 240.],[-426., 0.]], dtype=np.float32)
shape = (426, 240)
M = cv2.getPerspectiveTransform(points_src, points_dst)
warped_gtimg = cv2.warpPerspective(gt_img, M, shape, cv2.INTER_LINEAR, cv2.BORDER_CONSTANT, 0)
plt.imshow(warped_gtimg)

#################### tgm #####################
gt_img_ts = torch.from_numpy(gt_img).unsqueeze(0).unsqueeze(0)
points_src_ts = torch.from_numpy(points_src).unsqueeze(0)
points_dst_ts = torch.from_numpy(points_dst).unsqueeze(0)
M = tgm.get_perspective_transform(points_src_ts, points_dst_ts)
img_warp = tgm.warp_perspective(gt_img_ts.float(), M, dsize=(240, 426))
print(img_warp.size())
np_warped_gtimg = img_warp.cpu().numpy()[0][0]
plt.imshow(np_warped_gtimg)
```

**Comment by edgarriba** [comment]
@Msabih thanks. I'll investigate that

**Comment by Msabih** [comment]
@edgarriba  were you able to reproduce the issue ?

**Comment by edgarriba** [comment]
@Msabih yes, I run your script and I get an empty image.

**Comment by Msabih** [comment]
@edgarriba  and the opencv gives lines correctly. I will also try to debug this. Any hints will be welcome. Thanks for your time.

**Comment by jiangwei221** [comment]
@Msabih I'd like to reproduce the bug, but the provided .npz file is expired

**Comment by edgarriba** [comment]
@jiangwei221 thanks to help with that, I'm still working on infrastructure stuff and couldn't spend time on this. I can still download the file.

**Comment by jiangwei221** [comment]
Caused by #146 .

**Comment by bhack** [comment]
Is it solved?

**Comment by edgarriba** [comment]
@jiangwei221 @Msabih can we close this ?

**Comment by edgarriba** [comment]
I think this is solved. Closing.
